tools/report-converter/codechecker_report_converter/cli.py

```diff
@@ -137,15 +121,17 @@ def main():
     if 'verbose' in args and args.verbose:
         LOG.setLevel(logging.DEBUG)
 
+    clean = True if 'clean' in args else False
+
     if 'input' in args and args.input:
-        with open(args.input) as input_file:
-            output = input_file.readlines()
-    else:
-        output = sys.stdin.readlines()
+        return output_to_plist(args.input, args.type, args.output_dir, clean)
 
-    clean = True if 'clean' in args else False
+    with tempfile.NamedTemporaryFile() as analyzer_result_file:
```

Comment:
why do we need this NamedTemporaryFile?
Is the output_to_plist called twice?

Reply:
You can pipe the result of an analyzer tool to this converter tool:
`clang-tidy ... | report-converter ...`

The above function `output_to_plist` requires a file path so this is the reason why we create a temporary file here.

Reply:
I think we should highlight this usage for each analyzer, because not every type of output conversion is right this way, can we parse the cppcheck output with this pipe method? It can be a bit confusing because it can be a file/directory/stream but not every option is supported for all of the analyzers.
This is really analyzer dependent, probably piping will work to the sanitizers too but not for findbugs, infer ...